libxw/src/libxw/puzzle.py

```diff
@@ -41,9 +42,39 @@ class BlackSquare:
     pass
 
 
+_POSITIVE_INFINITY = float("inf")
+_NEGATIVE_INFINITY = float("-inf")
+
+
 @dataclass
 class Puzzle:
     BLACK_SQUARE = BlackSquare()
 
     cells: Mapping[CellIndex, Union[Cell, BlackSquare]]
     clues: Mapping[ClueDirection, Mapping[CellIndex, Clue]]
+
+    @property
+    def width(self) -> int:
```

Comment:
These methods are really nasty.  Once we have a better idea of the APIs we need to support, we can swap these out for something faster.